pageserver/src/tenant/timeline.rs

```diff
@@ -2264,7 +2264,15 @@ impl Timeline {
                         )));
                     }
                 }
-                ancestor.wait_lsn(timeline.ancestor_lsn, ctx).await?;
+                ancestor
+                    .wait_lsn(timeline.ancestor_lsn, ctx)
+                    .await
+                    .with_context(|| {
+                        format!(
+                            "failed to wait for lsn {} on ancestor timeline_id={}",
```

Comment:
nit: useless "failed to" prefix. Only use it when logging the error, but not in .context() or .with_context(). Makes up for a nicer and more concise `context: context: context` chain

Reply:
Good point, of course.

Reply:
[71b16b4](https://github.com/neondatabase/neon/pull/5143/commits/71b16b4e6a43cd9d680950eac8cd9778afb71065)